File: Ruby_version/Napakalaki/lib/prize.rb
# encoding: utf-8
# being prize.rb

# Clase Prize que representa la recompensa por vencer a un monstruo
module NapakalakiGame
  class Prize

    # Consultores / datos miembro
    # -------------------------------------------------------

    #attr_reader :treasures  # Nº de tesoros a coger
    #attr_reader :levels      # Nº de niveles aumentados

    # -------------------------------------------------------

    # Constructor
    # -------------------------------------------------------

    def initialize(t, l)
      @treasures = t
      @levels = l
    end

    # -------------------------------------------------------

    # Métodos
    # -------------------------------------------------------

    # Convierte a string
    def to_s
      "\nTesoros: #{@treasures}\nNiveles: #{@levels}"
    end

    def getTreasures
      @treasures
    end

    def getLevels
      @getLevels
    end

    # -------------------------------------------------------
  end
end

# end prize.rb

File: Java_version/Napakalaki/src/napakalaki/Prize.java
// begin Prize.java
package napakalaki;

// Clase Prize que representa la recompensa a obtener
public class Prize {
    
    /* Datos miembro */
    /* ------------------------------------------------------ */
    
    // Nº de tesoros obtenidos
    private int treasures;
    // Nº de niveles subidos
    private int levels;
    
    
    public Prize(int t, int l) {
        this.treasures = t;
        this.levels = l;
    }
   
    
    public int getTreasures() {
        return treasures;
    }
    
    public int getLevels() {
        return levels;
    }

    public String toString() {
        return  "\n Tesoros = " + Integer.toString(treasures) 
                + "\n Niveles = " + Integer.toString(levels);
    }
    
    /* ------------------------------------------------------ */
}

// end Prize.java